src/Services/PatientService.php

```diff
@@ -244,6 +244,16 @@ public function getAll($search = array())
                 if (in_array($fieldName, array('fname', 'lname', 'street'))) {
                     array_push($whereClauses, $fieldName . ' LIKE ?');
                     array_push($sqlBindArray, '%' . $fieldValue . '%');
+                } elseif ($fieldName === "name") {
```

Comment:
The search terms/fields within the OpenEMR Patient API seems to be aligned with the patient_data column names. My previous PR attempted to follow the spirit of that approach.

I added a constant to limit the supported search fields in the [Patient Rest Controller](https://github.com/openemr/openemr/blob/master/src/RestControllers/PatientRestController.php#L25) so that we would all have a better idea of the support terms. The choices I made were to better support the upcoming work I'm doing for the Patient FHIR R4 API.

This particular conditional code block attempts to simplify search by noting that a handful of terms utilize wildcard/like expressions, while the remaining terms utilize an exact match strategy. 
```php
foreach ($search as $fieldName => $fieldValue) {
   if (term is a wildcard term) {
      <build a like clause/predicate>
   } else {
      <do an exact match>
   }
}
```
My intent  is to support specific searches by an API client, just to keep things clean and tidy and simplify the queries which are sent to the database server.

So a search for "name" would require a search for first name and last name. This is in the whitelist of fields on line 244 so a wild card search will be used.
```php
/api/patient?fname=John&lname=Doe
```

Each address field is supported separately rather than providing an "address" parameter which is OR'd.
```php
api/patient?street=500 Cherry Lane&postal_code=90210&etc=etc
```
Of course this is just my viewpoint. If some of the core committers, @sjpadgett  or @stephenwaite , feel that the API search parameters should be adjusted/tweaked a bit, then we can do so in a way that makes the most sense. For example we could delegate some of the fancier parameter building into a separate function or two so that we don't have a growing conditional statement. 

We can see what sort of feedback we receive and take it from there. 

Thank you very much for reviewing these recent changes!

Reply:
Thanks for explaining so well. I truly agree with you that things should be clean and tidy and as It is Core Patient API it doesn't need those search parameters. So I have come up with a better solution and rather than changing core Patient I tweaked the FHIR controller.

Reply:
No problem @yashrajbothra. If you're looking into the FHIR controller you will need to be aware of the [support search parameters](https://www.hl7.org/fhir/patient.html#searchl). OpenEMR won't support all of these parameters, but rather a subset of what is actually maintained in the patient_data table and what the core team would like to support.

Reply:
Hey, @sjpadgett @rhausam As our database doesn't store the whole address and just support its subsets so should we remove the `address` search parameter. If not what should happen if somebody uses both search parameters like `address` and its subsets. I tried finding an answer on spec and testing on HAPI. 
![image](https://user-images.githubusercontent.com/47683556/80394649-c0a70300-88cf-11ea-9f0c-af78560a9368.png)
https://www.hl7.org/fhir/search.html#string